test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/IdentityHashMapTest.scala

```diff
@@ -0,0 +1,120 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+package org.scalajs.testsuite.javalib.util
+
+import java.{util => ju}
+import scala.reflect.ClassTag
+import org.junit.Assert._
+import org.junit.Test
+
+class IdentityHashMapTest extends MapTest {
+  override def factory: IdentityHashMapFactory = new IdentityHashMapFactory
+
+  // tests from Harmony
```

Comment:
It seems everything here is already covered by tests we have in `MapTest`. It doesn't seem worth repeating them.

However what is missing here are tests for reference equality semantics, which is the whole point of `IdentityHashMap`.

Reply:
I added `def allowsIdentityBasedKeys: Boolean = false` to `MapFactory` and had an `if` on `TestObj`. This is how I did it on Scala Native. Also, since Strings can be interned on JVM, the references are the same. Scala Native there seems to cache numbers as well so the references are the same and the tests work.

I can't get all the current tests working yet so not sure what I need to do yet. Not sure what is wrong yet.

I am also having a link problem when I try and run 
`testSuite/testOnly *IdentityHashMap*`
I get `[error] Fatal error: java.lang.Object is missing`

Reply:
I would be happy to supply good tests once I get something working ok and the current tests pass.